File: BAEKJOON/2212 - 센서/센서.py
def min_total_distance(N, K, sensors):
    # 집중국의 수가 센서의 개수 이상이면 모든 센서를 개별적으로 커버할 수 있으므로 거리 합은 0
    if K >= N:
        return 0

    # 센서 좌표를 오름차순으로 정렬하여 계산을 간단하게 만듦
    sensors.sort()

    # 인접한 센서들 간의 거리 차이를 저장할 리스트 생성
    distances = []
    for i in range(1, N):
        # 각 센서 간의 거리를 계산하여 distances 리스트에 추가
        distances.append(sensors[i] - sensors[i - 1])

    # 거리를 내림차순으로 정렬하여 가장 큰 간격부터 처리 가능하게 함
    distances.sort(reverse=True)

    # 집중국이 K개라면, 가장 큰 K-1개의 간격을 제거
    for _ in range(K - 1):
        distances.pop(0)

    # 남은 간격의 합을 반환하여 최종 최소 거리 합을 계산
    return sum(distances)


# 첫 번째 입력: 센서의 개수
N = int(input().strip())

# 두 번째 입력: 집중국의 개수
K = int(input().strip())

# 세 번째 입력: 센서의 좌표들
sensors = list(map(int, input().strip().split()))

# 최종 계산 결과를 출력
print(min_total_distance(N, K, sensors))

File: BAEKJOON/2212 - 센서/센서.cs
using System;
using System.Collections.Generic;

class Program {
    // 최소 수신 가능 거리의 합을 계산하는 함수
    static int MinTotalDistance(int N, int K, List<int> sensors) {
        // 집중국의 수가 센서의 수 이상인 경우, 모든 센서를 개별적으로 커버 가능
        if (K >= N) return 0;

        // 센서 좌표를 오름차순으로 정렬
        sensors.Sort();

        // 인접 센서 간의 거리를 저장할 리스트
        List<int> distances = new List<int>();
        for (int i = 1; i < N; i++) {
            // 센서 간 거리 계산
            distances.Add(sensors[i] - sensors[i - 1]);
        }

        // 거리 리스트를 내림차순으로 정렬
        distances.Sort((a, b) => b.CompareTo(a));

        // 가장 큰 K-1 개의 간격을 제거
        for (int i = 0; i < K - 1; i++) {
            distances[i] = 0; // 큰 간격 제거
        }

        // 남은 거리의 합을 계산
        int result = 0;
        foreach (int distance in distances) {
            result += distance;
        }
        return result;
    }

    static void Main(string[] args) {
        // 입력 처리
        int N = int.Parse(Console.ReadLine());
        int K = int.Parse(Console.ReadLine());

        string[] input = Console.ReadLine().Split();
        List<int> sensors = new List<int>();
        foreach (string s in input) {
            sensors.Add(int.Parse(s));
        }

        // 최소 거리 합 계산 및 출력
        Console.WriteLine(MinTotalDistance(N, K, sensors));
    }
}